Expensify/App
# Add a modal screen for creating a new workspace

### Details
Adds a new modal screen for creating workspaces (UI only this is not connected to the backend yet). Only accessible through the web with the path `/workspace/new` for now.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/166035

### Tests
1. Open E.cash and log in.
2. Change the URL to http://localhost:8080/workspace/new
3. Make sure everything looks like it should.

### QA Steps
1. Open E.cash and log in.
2. Change the URL to https://staging.expensify.cash/workspace/new
3. Make sure everything looks like it should.

### Tested On


### Screenshots


#### Web
![image](https://user-images.githubusercontent.com/6759612/121709301-b39ba200-cad8-11eb-98ce-96266549a5dc.png)

#### Mobile Web
![image](https://user-images.githubusercontent.com/6759612/121709936-5bb16b00-cad9-11eb-9282-f8771e9a67f5.png)

#### Desktop
![image](https://user-images.githubusercontent.com/6759612/121717525-90292500-cae1-11eb-82be-0fe802f02549.png)

#### iOS
![image](https://user-images.githubusercontent.com/6759612/121716392-42f88380-cae0-11eb-8767-04a8de800e88.png)

#### Android
![image](https://user-images.githubusercontent.com/6759612/121716997-024d3a00-cae1-11eb-8bea-09ec57748bf7.png)
